server/__tests__/words.test.js

```diff
@@ -0,0 +1,143 @@
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body).toHaveLength(0);
+    });
+});
+
+// update a single word
+describe('PATCH /words/word', () => {
+    it('on word matched, should return 1 modifiedCount with an http status 200', async () => {
+        const payload = { word: "test", hint: "test1", newHints: ["1", "2", "3"] };
+        const res = await supertest(server)
+            .patch('/words/word')
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body.acknowledged).toBe(true);
+        expect(res.body.modifiedCount).toBe(1);
+    });
+
+    it('on word not found, should return 0 modifiedCount with an http status 200', async () => {
+        const payload = { word: "no word", hint: "test1", newHints: ["1", "2", "3"] };
+        const res = await supertest(server)
+            .patch('/words/word')
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body.acknowledged).toBe(true);
+        expect(res.body.modifiedCount).toBe(0);
+    });
+
+    it('on wrong request body, should return false acknowledgement with an http status 200', async () => {
+        const payload = { word: "no word", hint: "test1" };
+        const res = await supertest(server)
+            .patch('/words/word')
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body.acknowledged).toBe(false);
+    });
+});
+
+// delete a single word
+describe('DELETE /words/word', () => {
+    it('on word matched, should return 1 deletedCount with an http status 200', async () => {
+        const payload = { word: "test", hint: "test1" };
+        const res = await supertest(server)
+            .delete('/words/word')
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body.acknowledged).toBe(true);
+        expect(res.body.deletedCount).toBe(1);
+    });
+
+    it('on word not found, should return 0 deletedCount with an http status 200', async () => {
+        const payload = { word: "no word", hint: "test1" };
+        const res = await supertest(server)
+            .delete('/words/word')
+            .send(payload)
+            .set('Content-Type', 'application/json');
+        expect(res.status).toEqual(200);
+        expect(res.body.acknowledged).toBe(true);
+        expect(res.body.deletedCount).toBe(0);
+    });
+});
+
+// create a multiple words
+describe('POST /words', () => {
+    it('on success, should return the created result with an http status 200', async () => {
+        const payload = {
+            words: [
+                { word: "my word1", hints: ["hint1", "hint2"] },
```

Comment:
Are we having multiple hints for a single word? How would showing the hint prior to a round starting work with multiple hints?

Reply:
I just made it a list for now since I thought we could potentially control the difficulty by providing more hints (or as a powerup?). Alyssa mentioned in class that we could have some sort of difficulty property (and maybe a category) for the words.

Reply:
Yeah that's smart, we'll need them if we want to make a category based on certain themes. More hints as a powerup sounds great too!